FaceDetectionServer/moedx/object_detection/object_detector.py

```diff
@@ -0,0 +1,189 @@
+# Copyright 2018-2020 MobiledgeX, Inc. All rights and licenses reserved.
+# MobiledgeX, Inc. 156 2nd Street #408, San Francisco, CA 94105
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# First off, add the submodule's path so we can import its packages.
+import os, sys
+submodule = "../pytorch_objectdetecttrack"
+sys.path.append(os.path.join(os.path.dirname(__file__), submodule))
+
+from models import *
+from utils import *
+
+import time, datetime, random
+import torch
+from torch.utils.data import DataLoader
+from torchvision import datasets, transforms
+from torch.autograd import Variable
+
+import matplotlib.pyplot as plt
+import matplotlib.patches as patches
+from PIL import Image
+
+package_dir = os.path.dirname(os.path.abspath(__file__))+"/"+submodule
+config_path=package_dir+'/config/yolov3.cfg'
+weights_path=package_dir+'/config/yolov3.weights'
+class_path=package_dir+'/config/coco.names'
+img_size=416
+conf_thres=0.8
+nms_thres=0.4
+
+# Load model and weights
+model = Darknet(config_path, img_size=img_size)
+model.load_weights(weights_path)
+
+if torch.cuda.is_available():
+    model.cuda()
+    Tensor = torch.cuda.FloatTensor
+else:
+    Tensor = torch.FloatTensor
+
+model.eval()
+classes = utils.load_classes(class_path)
+
+# Globals
+total_server_processing_time = 0
+count_server_processing_time = 0
+
+class ObjectDetector(object):
+    def __init__(self):
+        pass
+
+    def detect_image(self, img):
+        # scale and pad image
+        ratio = min(img_size/img.size[0], img_size/img.size[1])
+        imw = round(img.size[0] * ratio)
+        imh = round(img.size[1] * ratio)
+        img_transforms = transforms.Compose([ transforms.Resize((imh, imw)),
+             transforms.Pad((max(int((imh-imw)/2),0), max(int((imw-imh)/2),0), max(int((imh-imw)/2),0), max(int((imw-imh)/2),0)),
+                            (128,128,128)),
```

Comment:
What is (128, 128, 128)? A color?

Reply:
Yes. When the image is resized, it builds a gray border around it to maintain aspect ratio of the original image.